module/tools/RoboCupConfiguration/src/RoboCupConfiguration.cpp

```diff
@@ -0,0 +1,489 @@
+        if (get_platform() != "nugus") {
+            log_message = "Configure Error: Network configuration only available on NUgus robots!";
+            return;
+        }
+
+        // Check if any of ip_address, ssid or password are empty
+        if (ip_address.empty() || ssid.empty() || password.empty()) {
+            log_message = "Configure Error: IP Address, SSID and Password must be set!";
+            return;
+        }
+
+        /* WIRELESS NETWORK SYSTEMD CONFIG */
+
+        // Get folder name
+        const std::string systemd_folder = fmt::format("system/{}/etc/systemd/network", hostname);
+
+        // Parse the IP address
+        std::stringstream ss(ip_address);
+        std::vector<std::string> ip_parts{};
+        for (std::string part; std::getline(ss, part, '.');) {
+            ip_parts.push_back(part);
+        }
+
+        // Check if team_id and player_id match with third and fourth parts of the IP address.
+        // This shouldn't break things, but the user should know that there may be an issue.
+        // In the lab, the IP addresses of the robots are set and may not match in this way,
+        // but at RoboCup, it is important that this format is followed
+        if (team_id != std::stoi(ip_parts[2]) || player_id != std::stoi(ip_parts[3])) {
+            log_message +=
+                "Warning: At RoboCup, the third position of the IP address should be the team ID and the fourth should "
+                "be the player number. ";
+        }
+
+        // Write the new ip address to the file
+        std::ofstream(fmt::format("{}/30-wifi.network", systemd_folder))
+            << fmt::format("[Match]\nName={}\n\n[Network]\nAddress={}/16\nGateway={}.{}.3.1\nDNS=8.8.8.8",
+                           wifi_interface,
+                           ip_address,
+                           ip_parts[0],
+                           ip_parts[1]);
+
+
+        /* WPA_SUPPLICANT CONFIG */
+
+        // Make a new wpa_supplicant file in the robot-specific directory
+        // This is so that we don't lose the original if we need it
+        // And so when we append to it with a high priority, we are doing it from fresh
+        // And will not encounter any issues when running multiple times
+        std::string wpa_supplicant_dir  = fmt::format("system/{}/etc/wpa_supplicant", hostname);
+        std::string wpa_supplicant_file = fmt::format("{}/wpa_supplicant-{}.conf", wpa_supplicant_dir, wifi_interface);
+        system(("mkdir -p " + wpa_supplicant_dir).c_str());
+
+        std::filesystem::copy_file(
+            fmt::format("system/default/etc/wpa_supplicant/wpa_supplicant-{}.conf", wifi_interface),
+            wpa_supplicant_file,
+            std::filesystem::copy_options::overwrite_existing);
+
+        // Append to the wpa_supplicant file with a high priority
+        std::string command =
+            "wpa_passphrase " + ssid + " " + password + " | sed 's/}/\tpriority=9999\\n}/' >> " + wpa_supplicant_file;
```

Comment:
So this priority thing is yikes but I'm not sure how to make it nicer. Maybe parse the file and find the highest priority number. Practically this is probably fine I doubt anyone is setting a higher priority than this but still.

Reply:
Probs fine